src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -234,39 +206,24 @@ void RunCKSolution(const Handle& handle,
 }
 
 template <typename DataType>
-InvokerFactory MakeInvokerFactoryHelper(CKArgs ck_args, size_t config_idx)
+InvokerFactory MakeInvokerFactoryHipImplGemmFwdXdlops(CKArgs ck_args, size_t config_idx)
 {
-    return [ck_args = std::move(ck_args), config_idx](const std::vector<Kernel>& kernels) mutable {
-        std::ignore = kernels;
+    auto conv_ptrs = DeviceOpPtrs<DataType>::GetInstances();
+    auto conv_ptr  = std::move(conv_ptrs.at(config_idx));
```

Comment:
Are you sure you should move here? That invalidates the object in the map, while not removing it from there.

Reply:
1) It must be moved since it's a `std::unique_ptr`
2) It a prefect world `conv_ptrs` should not contain any "invalid" pointers
3) but we hold and index in this array, so we can't even remove any pointers from it at all
4) and it seems it doesn't matter. `DeviceOpPtrs<DataType>::GetInstances();` returns vector by value, so `conv_ptrs` will be destroyed anyway and it's not critical that we have one invalid entry.

But this particular construction is still error-prone. It's better to get rid of `conv_ptrs` at all.

Reply:
Done

Reply:
What @CAHEK7 said: i.e., we own the vector so we can and must do `move` because it's a `move-only` object. 
I have noticed that other solvers `search` through the `conv_ptrs` vector by string name, so this logic will most likely change. I think it is currently broken, right @iq136boy? (Please correct me if I'm wrong).